aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -773,6 +773,8 @@ module aptos_token::token {
         assert_tokendata_exists(creator, token_data_id);
         let all_token_data = &mut borrow_global_mut<Collections>(token_data_id.creator).token_data;
         let token_data = table::borrow_mut(all_token_data, token_data_id);
+        // cannot change maximum from 0 and cannot change maximum to 0
+        assert!(token_data.maximum != 0 && maximum != 0, error::invalid_argument(EINVALID_MAXIMUM));
```

Comment:
IIUC, please also add this line before line 1072, otherwise the invariant can not be guaranteed at initialization.

Reply:
Are you referring to this line? 
https://github.com/aptos-labs/aptos-core/blob/7117d12b9bc48f05771cf8bc52e9a219dccda5d9/aptos-move/framework/aptos-token/sources/token.move#L1072
IIUC, it is valid to have `token_data.maximum = 0`. The case here is to prevent mutating maximum from or to zero because when the maximum is zero, the supply is untracked. For example,
https://github.com/aptos-labs/aptos-core/blob/7117d12b9bc48f05771cf8bc52e9a219dccda5d9/aptos-move/framework/aptos-token/sources/token.move#L1196

Reply:
So it is allowed to have `maximum = 0` at initialization and immutable since then. n00b Q: Do we have any use case for that? cc @areshand

Reply:
`So it is allowed to have maximum = 0 at initialization and immutable since then. ` Nope, that's what this PR hopes to fix. The only usecase that should be allowed is : having `maximum > 0` at initialization, and then mutate it to a valid (>supply) non-zero value since then.